src/dso_api/dynamic_api/filters/parser.py

```diff
@@ -323,11 +321,23 @@ def _translate_lookup(
                     }
                 ) from None
 
+        # Handle case-insensitive exact matches for string fields only,
+        # but not for relations or formatted fields
         if filter_part.field.format == "date-time" and not isinstance(value, datetime):
             # When something different then a full datetime is given, only compare dates.
             # Otherwise, the "lte" comparison happens against 00:00:00.000 of that date,
             # instead of anything that includes that day itself.
-            lookup = f"date__{lookup or 'exact'}"
+            return f"date__{lookup or 'exact'}"
+
+        # Only apply iexact for direct string field lookups (not through relations)
+        if (
+            not lookup
+            and filter_part.field.type == "string"
+            and not filter_part.field.format  # Exclude formatted fields like date-time
+            and not filter_part.field.relation  # Exclude relation fields
```

Comment:
```suggestion
            and not filter_part.field.is_relation  # Exclude relation fields
```

Omdat `relation` op 1-n iets teruggeeft, en de `nm_relation` variatie negeert

Reply:
Changed